odoo/cli/tsconfig.py

```diff
@@ -0,0 +1,129 @@
+import argparse
+import glob
+import itertools
+import os
+import sys
+import re
+import json
+
+from . import Command
+from odoo.modules.module import MANIFEST_NAMES
+
+
+class TSConfig(Command):
+    """Generates tsconfig files for javascript code"""
+
+    def __init__(self):
+        self.command_name = "tsconfig"
+
+    def get_module_list(self, path):
+        mods = itertools.chain.from_iterable(
+            glob.glob(os.path.join(path, '*/%s' % mname))
+            for mname in MANIFEST_NAMES
+        )
+        return [mod.split(os.path.sep)[-2] for mod in mods]
+
+    def clean_path(self, path):
+        regex = r"//+"
+        subst = "/"
+        return re.sub(regex, subst, path)
+
+    def prefix_suffix_path(self, path, prefix, suffix):
+        return self.clean_path(f"{prefix}/{path}/{suffix}")
+
+    def remove_(self, modules, module):
+        for name, path in modules:
+            if module == name:
+                modules.remove((name, path))
+
+    def run(self, cmdargs):
+        parser = argparse.ArgumentParser(
+            prog="%s %s" % (sys.argv[0].split(os.path.sep)[-1], self.command_name),
+            description=self.__doc__
+        )
+        parser.add_argument('--addons-path', type=str, nargs='+', dest="paths")
+        args = parser.parse_args(args=cmdargs)
+
+        modules = {}
+        for path in args.paths:
+            for module in self.get_module_list(self.clean_path(path)):
+                modules[module] = self.prefix_suffix_path(module, path, "/static/src/*")
+
+        content = self.generate_file_content(modules)
+        print(json.dumps(content, indent=2))
```

Comment:
please add `# pylint: disable=bad-builtin` if you want to keep the print. (if I'm correct)